src/main/java/com/google/devtools/build/lib/actions/CommandLines.java

```diff
@@ -212,15 +211,9 @@ ExpandedCommandLines expand(
    */
   public ImmutableList<String> allArguments()
       throws CommandLineExpansionException, InterruptedException {
-    return allArguments(PathMapper.NOOP);
-  }
-
-  /** Variation of {@link #allArguments()} that supports output path stripping. */
-  public ImmutableList<String> allArguments(PathMapper stripPaths)
-      throws CommandLineExpansionException, InterruptedException {
     ImmutableList.Builder<String> arguments = ImmutableList.builder();
     for (CommandLineAndParamFileInfo pair : getCommandLines()) {
-      arguments.addAll(pair.commandLine.arguments(/*artifactExpander=*/ null, stripPaths));
+      arguments.addAll(pair.commandLine.arguments(/*artifactExpander=*/ null, PathMapper.NOOP));
```

Comment:
For the test, see https://gist.github.com/gregestren/cfcdd55a7620b6c6ce3d5e40f633ce88.

Reply:
An alternative to `aquery` could be to run a `bazel clean` in the beginning of the test and then use `--subcommands`. Mid-term `aquery` won't work as `PathMapper`s could use the hashes of input files to form paths, but `aquery` doesn't have access to this information.

Reply:
I vaguely recall `--subcommands` having problems seeing the same paths as the actual action. Maybe that's not an issue with your latest refactorings? I think this is another case of trying to follow which callers call which variations of arguments expansion.

We could also consider a Java integration test that intercepts the logic just before it goes to an actual executor. That might require a mock executor to work generically. But it could offer universal testing regardless of executor. Then use shell tests for black-box execution coverage, i.e "expect build B works and don't care how the logic is implemented". 

What would the story for full Bazel test support look like? We've been leaning toward excluding local execution from path mapping because it's not fundamentally compatible with the feature. But some variations of local execution *are* compatible, and I believe your intention is that your code naturally works with those implementations? Or would it be easier to support in-test RBE calls than I imagine? Or should we make a generic mock executor? I worry a mock would be complicated since part of "build success" involves knowing Java, etc. compilation runs correctly.

Reply:
Aside from details there's also process:

I'm not thrilled about it, but we could drop existing testing for the purpose of landing your refactoring. Then build new testing as a hopefully quick followup. Most of @aranguyen 's current testing is running standard CI with a custom Bazel that uses path mapping. She can still do that even if direct tests for this feature are disabled.